File: back-end/ponton/src/main/java/com/example/demo/entities/Comment.java
package com.example.demo.entities;

import lombok.AccessLevel;
import lombok.Data;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private Integer commentId;

    @ManyToOne(cascade = {CascadeType.MERGE}, fetch = FetchType.LAZY)
    @JoinColumn(name = "comments_users")
    private User user;

    @ManyToOne(cascade = {CascadeType.MERGE}, fetch = FetchType.LAZY)
    @JoinColumn(name = "comments_posts")
    private Post post;

    private String text;

    @CreationTimestamp
    @Setter(AccessLevel.NONE)
    private Date createDate;

    @UpdateTimestamp
    @Setter(AccessLevel.NONE)
    private Date modifiedDate;
}

File: front-end/frontBlog/src/app/entities/Comment.ts
import { User } from './User';
import { Post } from './Post';

export class Comments{
    commentId: Number;
    user: User;
    post: Post;
    text: string;
    createDate: Date;
    modifiedDate: Date;

    constructor(commentId: Number, user: User, post: Post, text: string, createDate: Date, modifiedDate: Date){
        this.commentId = commentId;
        this.user = user;
        this.post = post;
        this.text = text;
        this.createDate = createDate;
        this.modifiedDate = modifiedDate;
    }
}